quickannotator/db/colors.py

```diff
@@ -0,0 +1,21 @@
+from quickannotator.db.crud.annotation_class import search_annotation_class_by_project_id
+import quickannotator.constants as constants
+
+class ColorPalette():
+    def __init__(self, project_id):
+        color_palette_name = 'default'  # TODO: get this from project settings
+
+        if color_palette_name not in constants.ANNOTATION_CLASS_COLOR_PALETTES:
+            raise KeyError(f"Color palette '{color_palette_name}' does not exist in constants.COLOR_PALETTES.")
+        
+        self.color_list = constants.ANNOTATION_CLASS_COLOR_PALETTES[color_palette_name]
+        self.project_id = project_id
+
+    def get_unused_color(self):
+        annotation_classes = search_annotation_class_by_project_id(self.project_id)
+        used_colors = {str(ac.color) for ac in annotation_classes}
```

Comment:
only minor comment looking at this section - its not obvious to me if this is always going to be determinsitic or not, especially since "set" makes no gaurantee about ordering?  its not a requirement, but its a "nice to have" that e.g., the first class is always color C1, the second class is always C2, etc - so that when going interproject there is some constancy